examples/models/llama2/quantize.py

```diff
@@ -131,507 +130,6 @@ def dynamically_quantize_per_channel(
-    to map from floating point to quantized values. This means for each row of a 2-d Tensor
-    (M, N), we calculate scales/zero_points for each `group_size` elements
-    and quantize every `group_size` elements with the same quantization parameter.
-    The dimension for scales/zero_points will be (M * ceil(N, group_size),)
-
-    Args:
-       input (torch.Tensor): quantized Tensor (uint8/int8 etc.)
-       scales (float32 torch.Tensor): quantization parameter for per channel group affine quantization
-       zero_points (int32 torch.Tensor): quantization parameter for per channel group affine quantization
-       quant_min (int): minimum quantized value for input Tensor
-       quant_max (int): maximum quantized value for input Tensor
-       dtype (torch.dtype): dtype (e.g. torch.uint8) for input Tensor
-       output_dtype (torch.dtype): dtype (e.g. torch.float32) for output Tensor
-
-    Returns:
-       dequantized Tensor with dtype `output_dtype`
-    """
-
-    assert group_size > 1
-    # needed for GPTQ single column dequantize
-    if group_size > w_int8.shape[-1] and scales.shape[-1] == 1:
-        group_size = w_int8.shape[-1]
-    assert w_int8.shape[-1] % group_size == 0
-    assert w_int8.dim() == 2
-
-    w_int8_grouped = w_int8.reshape(-1, group_size)
-    scales = scales.reshape(-1, 1)
-    zero_points = zero_points.reshape(-1, 1)
-    w_dq = (
-        w_int8_grouped.sub(zero_points).mul(scales).reshape_as(w_int8).to(output_dtype)
-    )
-    return w_dq
-
-
-def down_size(size):
-    assert size[-1] % 2 == 0, f"{size} last dim not divisible by two"
-    return (*size[:-1], size[-1] // 2)
-
-
-def up_size(size):
-    return (*size[:-1], size[-1] * 2)
-
-
-quantized_decomposed_lib.define("pack_int4_from_int8(Tensor int8_data) -> Tensor")
-
-
-@impl(quantized_decomposed_lib, "pack_int4_from_int8", "CompositeExplicitAutograd")
-def pack_int4_from_int8(int8_data: torch.Tensor) -> torch.Tensor:
-    # converting to uint8 for operations
-    shape = int8_data.shape
-    assert shape[-1] % 2 == 0
-    int8_data = int8_data.contiguous().view(-1)
-    return (int8_data[::2] << 4 | int8_data[1::2]).view(down_size(shape))
-
-
-quantized_decomposed_lib.define("unpack_int4_to_int8(Tensor int8_data) -> Tensor")
-
-
-@impl(quantized_decomposed_lib, "unpack_int4_to_int8", "CompositeExplicitAutograd")
-def unpack_int4_to_int8(int8_data: torch.Tensor) -> torch.Tensor:
-    """Get the original weight from the normalized float weight format"""
-    # since we are using int8 we will decode 2 entries per byte
-    # Shift elements down 4 and select out the bottom 4 bits
-    shape = int8_data.shape
-    first_elements = (int8_data >> 4).to(torch.int8)
-    second_elements = (int8_data & 0b1111).to(torch.int8)
-    return torch.stack([first_elements, second_elements], dim=-1).view(up_size(shape))
-
-
-def per_token_dynamic_quant(input: torch.Tensor) -> torch.Tensor:
```

Comment:
deleted here, but still used by linear_forward_8da4w

Reply:
we'll import from torchao